R/randomE.R

```diff
@@ -4,68 +4,208 @@
 #' distribution. Positive efficiencies indicate the consumption of resources,
 #' whilst negatives indicate that the species would produce the resource.
 #'
-#' @param n_species integer number of species
-#' @param n_resources integer number of resources
-#' @param min_con integer minimum number of resources consumed by each species
-#' @param max_con integer maximum number of resources consumed by each species
-#' @param min_prod integer minimum number of resources produced by each species
-#' @param max_prod integer maximum number of resources produced by each species
-#' @param maintenance numeric value between 0~1 the proportion of resources used
+#' @template man_spe
+#' @template man_res
+#' @param mean_consumption Numeric: mean number of resources consumed by each
+#' species drawn from a poisson distribution
+#' (default: \code{mean_consumption = n_resources/4})
+#' @param mean_production Numeric: mean number of resources produced by each
+#' species drawn from a poisson distribution
+#' (default: \code{mean_production = n_resources/6})
+#' @param maintenance Numeric: proportion of resources that cannot be converted
+#' into products
+#' between 0~1 the proportion of resources used
 #' to maintain the living of microorganisms. 0 means all the resources will be
 #' used for the reproduction of microorganisms, and 1 means all the resources
 #' would be used to maintain the living of organisms and no resources would be
 #' left for their growth(reproduction).
+#' (default: \code{maintenance = 0.5})
+#' @param trophic_levels Integer: number of species in microbial trophic levels.
+#' If NULL, by default, microbial trophic levels would not be considered.
+#' (default: \code{trophic_levels = NULL})
+#' @param trophic_preferences List: preferred resources and productions of each
```

Comment:
Consider using @inheritParams to use the same documentation in multiple functions and avoid a) repeating or b) providing different definitions for the same parameters.

Just add

@inheritParams myfunctionname 

and it will add descriptions those parameters that are already defined in the other function

Reply:
But perhaps @template is already enough. Not sure, just check what works best.

Reply:
@template is good enough, and these are not repeating with any other functions.